File: web/src/services/audit.ts
import { request } from '@/lib';
import type { API } from '@/typings';

export interface AuditLog {
  id: number;
  trace_id: string;
  user_id: number;
  api_key_id: number;
  model_name: string;
  request_summary: string;
  response_summary: string;
  prompt_tokens: number;
  completion_tokens: number;
  status_code: number;
  error_message: string;
  latency_ms: number;
  cost: number;
  ip_address: string;
  user_agent: string;
  created_at: string;
}

export interface AuditLogPage {
  list: AuditLog[];
  total: number;
  page: number;
  page_size: number;
}

export interface AuditLogSearchParams {
  page?: number;
  pageSize?: number;
  user_id?: number;
  model?: string;
  status?: string;
  start?: string;
  end?: string;
}

export const auditService = {
  async search(params: AuditLogSearchParams): Promise<{ list: AuditLog[]; total: number }> {
    const query = new URLSearchParams();
    if (params.page) query.set('page', String(params.page));
    if (params.pageSize) query.set('pageSize', String(params.pageSize));
    if (params.user_id) query.set('user_id', String(params.user_id));
    if (params.model) query.set('model', params.model);
    if (params.status) query.set('status', params.status);
    if (params.start) query.set('start', params.start);
    if (params.end) query.set('end', params.end);

    const res = await request.get<API.SingleResponse<AuditLogPage>>(`/audit/logs?${query.toString()}`);
    const data = res.data.data;
    return { list: data?.list ?? [], total: data?.total ?? 0 };
  },

  async fetchByTraceId(traceId: string): Promise<AuditLog[]> {
    const res = await request.get<API.SingleResponse<AuditLog[]>>(`/audit/logs/${traceId}`);
    return res.data.data ?? [];
  },
};

File: internal/model/audit.go
package model

import "time"

type AuditLog struct {
	ID              uint      `gorm:"primaryKey" json:"id"`
	TraceID         string    `gorm:"index;size:36;not null" json:"trace_id"`
	UserID          uint      `gorm:"index" json:"user_id"`
	APIKeyID        uint      `gorm:"index" json:"api_key_id"`
	ProviderID      uint      `json:"provider_id"`
	ModelName       string    `gorm:"size:128" json:"model_name"`
	RequestSummary  string    `gorm:"type:text" json:"request_summary"`
	ResponseSummary string    `gorm:"type:text" json:"response_summary"`
	PromptTokens    int       `json:"prompt_tokens"`
	CompletionTokens int     `json:"completion_tokens"`
	StatusCode      int       `json:"status_code"`
	ErrorMessage    string    `gorm:"size:1024" json:"error_message"`
	LatencyMs       int64     `json:"latency_ms"`
	Cost            float64   `json:"cost"`
	IPAddress       string    `gorm:"size:64" json:"ip_address"`
	UserAgent       string    `gorm:"size:512" json:"user_agent"`
	CreatedAt       time.Time `gorm:"index" json:"created_at"`
}
